tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2735,6 +2735,34 @@ class HLSLExternalSource : public ExternalSemaSource {
 
   HRESULT CombineDimensions(QualType leftType, QualType rightType, QualType *resultType);
 
+  clang::TypedefDecl *HLSLExternalSource::GetMatrixShorthandTypes(
```

Comment:
This method is being defined within the HLSLExternalSource class definition, there's no need for the "HLSLExternalSource::" prefixing the method name.  Same with the next method below.
Also, GetMatrixShorthandTypes should probably not be plural, change to: GetMatrixShorthandType.